[cetigerlily] iP

cetigerlily opened the pull request:
## 🌷 tigerlily to-do 🌷
> each day, try to cross off some to-do's - unknown

tigerlily to-do...
* eases your worries of the day 
* is very user-friendly

in it's _current_ version, tigerlily to-do offers:
- [x] adding deadlines to remember
- [x] setting up general to-dos
- [x] scheduling events
- [ ] reminders for upcoming deadlines (coming soon...)

want to get tigerlily to-do? simply follow these steps:
1. download it from [here](https://open.spotify.com/user/18o4w3o3inyaf47jsc95nseqq?si=aa1a4716020145e5) 
2. double-click it to launch
3. start adding your tasks and enjoy an easier daily life!

if you're also a java programmer, tigerlily to-do can be used to practice java
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
save

Commit:
update save

Commit:
dates and times

Commit:
update message

Commit:
merge branch level 7

Commit:
Merge branch 'branch-Level-8'

Commit:
merge branch level 8

Commit:
more OOP

Commit:
update

Commit:
Merge branch 'add-gradle-support'

Commit:
add JUnit tests

Commit:
add JavaDoc comments

Commit:
update

Commit:
Merge branch 'master' into branch-A-JavaDoc

Commit:
update

Bransonlj commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,70 @@
+
+public class Task {
+    private String description;
+    private boolean status;
```
maybe "status" could be renamed to sound more booleanish like isDone or isComplete!

Bransonlj commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,33 @@
+import duke.tasks.Task;
+
+public class DeleteCommand implements Command {
+    private String input;
```
maybe can add spaces between these lines!

Bransonlj commented on src/main/java/duke/exceptions/DukeExceptions.java:
```diff
@@ -0,0 +1,7 @@
+package duke.exceptions;
+
+public class DukeExceptions extends Exception {
```
Maybe can add javadoc for this and other classes to explain what it is for?

Bransonlj reviewed (commented):
Looks good, hard to find stuff to comment on

Commit:
fixing

Commit:
edit

Commit:
edit to follow coding standard

Commit:
implement find feature

Commit:
update exceptions

Commit:
merge level 9

Commit:
resolve merge conflicts

Commit:
add JavaDoc comments

Commit:
update

Commit:
merge JavaDoc

Commit:
update name

Cyrus-Krispin commented on src/main/java/tigerlily/tasks/Event.java:
```diff
@@ -0,0 +1,59 @@
+    public String toString() {
+        String timeFrame;
+        if ((eventStart.getYear() == eventEnd.getYear()) && (eventStart.getDayOfYear() == eventEnd.getDayOfYear())) {
+            timeFrame = eventStart.format(DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT)) + " " +
+                    eventStart.format(DateTimeFormatter.ofPattern("HH:mm")) + " to: " +
+                    eventEnd.format(DateTimeFormatter.ofPattern("HH:mm")) + ")";
```
Would it be better to have like breaks before the operator? I noticed this violation of coding standards in other places too.

Cyrus-Krispin commented on src/main/java/tigerlily/Parser.java:
```diff
@@ -0,0 +1,106 @@
+     * @return the required Command
+     * @throws UnknownCommandException If a given command doesn't match the supported command types
+     */
+    public Command handleCommand(String input) throws DukeExceptions {
+        try {
+            CommandType command = CommandType.valueOf(input.split(" ")[0].toUpperCase());
+            switch (command) {
+            case BYE:
+                return new ByeCommand();
+            case LIST:
+                return new ListCommand();
+            case TODO:
+                return new ToDoCommand(input);
+            case DEADLINE:
+                return new DeadlineCommand(input);
+            case EVENT:
+                return new EventCommand(input);
+            case MARK:
+                return new MarkCommand(input);
+            case UNMARK:
+                return new UnmarkCommand(input);
+            case DELETE:
+                return new DeleteCommand(input);
+            case FIND:
+                return new FindCommand(input);
+            default:
+                throw new UnknownCommandException();
+            }
+        } catch (IllegalArgumentException e) {
+            throw new UnknownCommandException();
+        }
+    }
```
I like the use of a switch rather than multiple if/nested-if statements for ur handleCommand method.

Cyrus-Krispin commented on src/main/java/tigerlily/commands/Command.java:
```diff
@@ -0,0 +1,10 @@
+package tigerlily.commands;
+
+import tigerlily.Storage;
+import tigerlily.Ui;
+import tigerlily.exceptions.DukeExceptions;
+import tigerlily.tasks.TaskList;
+
+public interface Command {
+    void execute(TaskList taskList, Ui ui, Storage storage) throws DukeExceptions;
+}
```
I like how a separate Command class is used to handle all the commands.

Cyrus-Krispin commented on src/main/java/tigerlily/Storage.java:
```diff
@@ -0,0 +1,74 @@
+     */
+    public void updateTasks(TaskList tasks) throws DukeExceptions {
+        try {
+            FileWriter fw = new FileWriter(this.filePath);
+            StringBuilder sb = new StringBuilder();
```
Is using more meaningful variable names a better option here?

Commit:
make edits based on PR feedback

Commit:
add GUI

Commit:
make more edits based on PR feedback

Commit:
merge branch level 10

Commit:
update build.gradle

Commit:
Add assertions

Assertions are not present in code.

Assertions are useful because they allow for the testing for validity of assumptions which were made in the program.

Let’s add assertions to Parser, TaskList, Mark Command, and Unmark Command.

Having assertions helps the smooth running of our code.

Commit:
Review code quality

Code can be further reviewed to adhere to proper code quality standards.

Code quality is important because it not only allows other users to read and understand it better, it will help you understand your own code better too.

Let's edit the existing code to follow proper code quality standards.

Having clean code allows us to understand our code better.

cetigerlily committed:
Merge pull request #2 from cetigerlily/branch-A-Assertions

Add assertions

Commit:
update

cetigerlily committed:
Merge pull request #4 from cetigerlily/branch-A-CodeQuality

Review code quality

Commit:
implement help feature

Commit:
add JavaDocs for command constructors

Commit:
merge branch-C-Help

Commit:
update for code quality

Commit:
improve GUI

Commit:
add new package

Commit:
merge branch-A-BetterGUI

Commit:
update README.md

Commit:
update README.md

Commit:
update for release

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	docs/Ui.png

Commit:
update README.md

Commit:
update README.md

cetigerlily committed:
Delete README.md
